substratevm/src/com.oracle.svm.agent.core/src/com/oracle/svm/agent/core/HandleSet.java

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright (c) 2019, 2020, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.  Oracle designates this
+ * particular file as subject to the "Classpath" exception as provided
+ * by Oracle in the LICENSE file that accompanied this code.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+package com.oracle.svm.agent.core;
+
+import static com.oracle.svm.core.util.VMError.guarantee;
+import static com.oracle.svm.jni.JNIObjectHandles.nullHandle;
+
+import java.util.Arrays;
+import java.util.concurrent.locks.ReentrantLock;
+
+import org.graalvm.nativeimage.c.type.CTypeConversion;
+
+import com.oracle.svm.jni.nativeapi.JNIEnvironment;
+import com.oracle.svm.jni.nativeapi.JNIMethodId;
+import com.oracle.svm.jni.nativeapi.JNIObjectHandle;
+
+public abstract class HandleSet {
```

Comment:
I find HandleSet is an overly generic name. Can we give a name to this set that describes all the handles well.

Reply:
How about JNIHandleSet?

Reply:
Would `BaseJVMTIHandleSet` or `JVMTIHandleSetBase` describe it well?

Reply:
At least having a javadoc comment for the class would help clarify what kind of handles this class manages.

Reply:
@vjovanov I can rename it to `JVMTIHandleSetBase`. The reason for the `JNIHandleSet` is because the class manages `JNI` such as `JNIObjectHandle`/`JNIMethodId`. It could also be `JNIHandleSetBase`.
@dougxc I agree, I will add documentation for the class.

Reply:
They are not JVMTI handles though (there is no such thing), these are a set of JNI handles frequently used by the agent, so I propose `AgentHandleSet`.